src/pyclass.rs

```diff
@@ -73,7 +75,7 @@ pub unsafe fn tp_free_fallback(obj: *mut ffi::PyObject) {
 /// The `#[pyclass]` attribute automatically implements this trait for your Rust struct,
 /// so you don't have to use this trait directly.
 pub trait PyClass:
-    PyTypeInfo<Layout = PyCell<Self>> + Sized + PyClassAlloc + PyMethodsImpl
+    PyTypeInfo<Layout = PyCell<Self>> + Sized + PyClassAlloc + PyMethodsImpl + PyProtoMethodsImpl
```

Comment:
I'm trying to understand - why is it necessary to make a new `PyProtoMethodsImpl` rather than using `PyMethodsImpl` for all of these?

Reply:
Technically we can reuse `PyMethodsImpl` but I felt it easier to understand to have two inventory types corresponding to `#[pymethods]` and `#[pyproto]`.
Not so string reason, though.

Reply:
As this isn't a user-facing API, I guess the only people who need to understand it are us. I probably have a preference for just one inventory type so that there's less code overall to understand, but happy if you prefer to keep it as two.

Reply:
I removed `PyProtoMethodsImpl`.